theano/scan_module/scan_opt.py

```diff
@@ -1964,8 +1964,10 @@ def scan_merge_inouts(node):
             outputs = [outputs]
 
         na = scan_args(outer_inputs, outputs, op.inputs, op.outputs, op.info)
+        remove = [node]
     else:
         na = a
+        remove = []
```

Comment:
remove can't be none when calling replace_all_validate_remove(). It must be an iterable. The extra work is so low that I don't think it is useful to make 2 code path to not pass the remove parameter.